### PR TITLE
Change profiling HTTP request runtime and type fields

### PR DESCRIPTION
**Pull request by flodav**
We want the profiling analyzer to process Ruby profiles. Today the quickest way is to set the `type` field to `auto` while waiting for some heavy refactoring into the analyzer.

We allow ourself to do this change since the profiler is still in beta.

### CODE REVIEW

**Comment by delner** [comment]
Made some updates to this:

 - Fixed tests for the `types[0] = 'auto'` change
 - Changed runtime tag to reflect language instead of interpreter (engine + platform)
 - Added runtime_engine and runtime_platform tags

These changes should address some needed things for the backend.

**Review by flodav — COMMENTED** [review]
LGTM but I can't approve my own PR. Maybe @jd you can take a look to review it ?

File: lib/ddtrace/profiling/flush.rb [hunk 1]
```diff
@@ -27,7 +28,8 @@ def initialize(*args)
         self.version = version || Datadog.configuration.version
         self.host = host || Datadog::Runtime::Socket.hostname
         self.language = language || Datadog::Runtime::Identity.lang
-        self.runtime = runtime || Datadog::Runtime::Identity.lang_interpreter
+        self.runtime_engine = runtime_engine || Datadog::Runtime::Identity.lang_engine
+        self.runtime_platform = runtime_platform || Datadog::Runtime::Identity.lang_platform
```
Comment (marcotc) [inline lib/ddtrace/profiling/flush.rb:32]: We might want to report these in the tracer payload as well. Nothing to do in this PR, just a reminder for our future selves.

**Review by marcotc — APPROVED** [review]
👍

File: lib/ddtrace/profiling/flush.rb (hunk 1 above)
Comment (delner) [reply lib/ddtrace/profiling/flush.rb:32]: Good idea. In the tracer payload I think we send them as "interpreter" which is both terms combined.

**Pull request merged**